### PR TITLE
bump `node-lts` to 18 & `node-current` to 19

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description
Node LTS is on Version 18 (check [here](https://nodejs.org/))

## Release Notes

```release-note
breaking: Node-LTS to `v18.12.0` (from Node-LTS `v16.18.0`) & Node-Current to `v19.0.0` (from Node-Current `v18.12.0`)
```

## Documentation

This should go with https://github.com/gitpod-io/website/pull/2883

### CODE REVIEW

**Comment by sagor999** [comment]
@gitpod-io/engineering-ide @gitpod-io/engineering-webapp any concerns with bumping node from v16 to v18 by default?

**Comment by geropl** [comment]
@sagor999 LGTM, we're pinning it [here](https://github.com/gitpod-io/gitpod/blob/3393f87cd07b10b13d4298c20d5f8abf894f06be/dev/image/Dockerfile#L133-L138).

**Comment by akosyakov** [comment]
> any concerns with bumping node from v16 to v18 by default?

breaking users, we should communicate it well through our channels with a strategy how to pin

**Comment by filiptronicek** [comment]
We could add a section like "upcoming changes to workspace images" to the changelog to give a date and guidance for pinning like Anton mentioned. Maybe just breaking on users without prior notice isn't ideal here. @Siddhant-K-code

**Comment by atduarte** [comment]
Related with two ongoing internal RFCs:
- [Workspace Images - Pinned Language Versions ](https://www.notion.so/gitpod/Workspace-Images-Pinned-Language-Versions-9d9483fb3dda409889d4fe0aea8b0a88)
- [Workspace Images - Release Channels](https://www.notion.so/gitpod/Workspace-Images-Release-Channels-4234c06f69cd4c9692603938a3d8a0ff)

**Comment by gengjiawen** [comment]
> > any concerns with bumping node from v16 to v18 by default?
> 
> breaking users, we should communicate it well through our channels with a strategy how to pin

TBH,  user should keep their app compatible with latest LTS or they should lock their Node.js version.
Gitpod should focus on deliver LTS with latest security fix and features. Currently default to 18 is the right thing todo.

**Comment by Siddhant-K-code** [comment]
Is there any plans to bump `node-lts` version further with a warning ⚠️ in changelogs (due for 2 last months)?

Or we can simply close this PR 😃

**Comment by Siddhant-K-code** [comment]
Related https://github.com/gitpod-io/gitpod/issues/16072

**Comment by gengjiawen** [comment]
> > > any concerns with bumping node from v16 to v18 by default?
> > 
> > 
> > breaking users, we should communicate it well through our channels with a strategy how to pin
> 
> TBH, user should keep their app compatible with latest LTS or they should lock their Node.js version. Gitpod should focus on deliver LTS with latest security fix and features. Currently default to 18 is the right thing todo.

@atduarte @akosyakov Anything to push this forward ?

Also, for the breaking users. Actually they can lock their node versions, since gitpod also provide nvm.

**Comment by akosyakov** [comment]
I agree that our base should be LTS in this case. Node users are quite adaptive. My original comment was that it should be a part of Changelog.

In long term we should have a process which makes it clear for users via docs and changelog, and if possible executed automatically.

**Comment by akosyakov** [comment]
@loujaybee maybe we should do @filiptronicek suggestion for this changelog: https://github.com/gitpod-io/workspace-images/pull/966#issuecomment-1293147489 too late?

**Comment by gengjiawen** [comment]
This became drawbacks since many project requires node 18, especially related to chatgpt tools.

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'chatgpt@3.5.1',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.19.0', npm: '8.19.3' }
npm WARN EBADENGINE }
```

**Comment by Siddhant-K-code** [comment]
We would gonna update this on 28th Feb ([announcement in changelog](https://www.gitpod.io/changelog/january-changelog#workspace-images-os-and-node-versions-update)). Till then you could follow [this guide](https://www.gitpod.io/docs/introduction/languages/javascript#updating-node-versions) (by using `nvm`)

**Review by kylos101 — APPROVED** [review]
Dear @Siddhant-K-code ,

You are a gentleman and a scholar, thank you so much!

This will ship next week as latest. 

This should also ship tomorrow, or Monday at the latest, as a timestamped tag.

File: chunks/lang-node/chunk.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 variants:
-  - name: "16"
-    args:
-      NODE_VERSION: 16.19.0
   - name: "18"
     args:
       NODE_VERSION: 18.13.0
+  - name: "19"
+    args:
+      NODE_VERSION: 19.0.0
```
Comment (kylos101) [inline chunks/lang-node/chunk.yaml:7]:
```suggestion
      NODE_VERSION: 19.17.0
```

File: chunks/lang-node/chunk.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 variants:
-  - name: "16"
-    args:
-      NODE_VERSION: 16.19.0
   - name: "18"
     args:
       NODE_VERSION: 18.13.0
```
Comment (kylos101) [inline chunks/lang-node/chunk.yaml:4]:
```suggestion
      NODE_VERSION: 18.14.2
```

**Review by kylos101 — CHANGES_REQUESTED** [review]
@Siddhant-K-code could I ask you to accept the suggested changes? I can merge after.

Alternatively, we can close this PR in favor of https://github.com/gitpod-io/workspace-images/pull/1017/commits/acbb0bcae61310414c4ae54f66ce7c2df492e793.

Let me know your preference?

**Comment by Siddhant-K-code** [comment]
> Alternatively, we can close this PR in favor of https://github.com/gitpod-io/workspace-images/commit/acbb0bcae61310414c4ae54f66ce7c2df492e793.

yeah, this makes more sense to release all at once. Closing this PR 🙏🏽 

Thanks ✨

**Pull request closed without merging**